src/sif/autocost.cc

```diff
@@ -353,8 +359,18 @@ AutoCost::AutoCost(const CostingOptions& costing_options, uint32_t access_mask)
   alley_factor_ = costing_options.alley_factor();
 
   // Preference to use highways. Is a value from 0 to 1
-  float use_highways_ = costing_options.use_highways();
-  highway_factor_ = 1.0f - use_highways_;
+  // Factor for highway use - use a non-linear factor with values at 0.5 being neutral (factor
+  // of 0). Values between 0.5 and 1 slowly decrease to a maximum of -0.125 (to slightly prefer
+  // highways) while values between 0.5 to 0 slowly increase to a maximum of kMaxHighwayBiasFactor
+  // to avoid/penalize highways.
+  float use_highways = costing_options.use_highways();
+  if (use_highways >= 0.5f) {
+    float f = (0.5f - use_highways);
+    highway_factor_ = f * f * f;
```

Comment:
dumb q: whats the reasoning for the math here (why the cube)?

Reply:
obviously because @gknisely only has 3 F's to give! badum-cha!

Reply:
copied from motorcycle costing.  we were seeing better results.  @dnesbitt61 originally wrote this logic.  Do you know why you did this?

Reply:
some secret sauce likely arrived at doing some route testing with motorcycles routes to see what gave results that seemed good. Luckily I commented my approach:
  // Factor for trail use - use a non-linear factor with values at 0.5 being neutral (factor
  // of 0). Values between 0.5 and 1 slowly decrease to a maximum of -0.125 (to slightly prefer
  // trails) while values between 0.5 to 0 slowly increase to a maximum of the surfact_factor_
  // to avoid/penalize trails.
Now this was applied to motorcycles where there were 2 different route case: first being for road motorcycling and second more adventure motorcyling. So I cannot vouch for its use in auto costing

Reply:
Yeah, I guess RAD tests will show if this is generating good results for auto or not (which I believe is good, right @gknisely ?)

Reply:
@gknisely Can you copy that code comment and modify if necessary to this code? I`m with @mandeepsandhu here :)